lib/galaxy/model/__init__.py

```diff
@@ -3226,7 +3226,7 @@ def to_dict(self, view='collection', expose_dataset_path=False):
         if hda.extended_metadata is not None:
             rval['extended_metadata'] = hda.extended_metadata.data
 
-        for name, spec in hda.metadata.spec.items():
+        for name in hda.metadata.spec.keys():
```

Comment:
Minor suggestion only:`for name in hda.metadata.spec:` (`keys()` not needed here, unless you want it to be more explicit?). Same for the other 2 cases below.

Reply:
Following the Zen of Python "Explicit is better than implicit", I think it makes it clearer we are dealing with a dict.